### PR TITLE
[WIP] started v2

### PR DESCRIPTION
**Pull request by mickaelandrieu**
- [x] Remove configuration file from library to cli app
- [x] Remove ``gush``
- [x] Make ``Yaml`` component optional
- [x] Introduced interfaces to allow customizations of system
- [x] Introduced loader for YAML (MUST work with actual packs)
- [x] Introduced loader for PHP arrays
- [ ] Introduced mapping files for Doctrine
- [ ] Introduced a real validator of question data
- [ ] Introduced a ``contribute.php`` file that allow to create questions in YAML through cli and/or web
- [ ] Added docs to explain How to contribute to decks and how to make his own loaders
- [x] Tests MUST pass
- [x] PHP7/Symfony3 only
- [x] Remove use of real Symfony pack (not needed anymore)
- [x] create ``1.x`` branch on github

### CODE REVIEW

**Comment by 20uf** [comment]
Hello,

Ha, you're going to introduce an interface.

A little more detail about it?

**Comment by mickaelandrieu** [comment]
I introduce many interfaces to allow people use their own implementations of Question, Answer, Set and Loader. I want to limit the maintenance and refuse to make my classes more complex than needed.

Note as of today I'm not sure of my architecture, I need to play with it and challenge how my library react to extensions and uses.

I'm open to any suggestions :)

File: tests/Certificationy/QuestionTest.php [hunk 1]
```diff
@@ -100,16 +99,16 @@ public function testAreCorrectAnswers()
      */
     public function testIsMultipleChoice()
     {
-        $multipleChoiceAnswers = array(
+        $multipleChoiceAnswers = new Answers([
```
Comment (mickaelandrieu) [inline tests/Certificationy/QuestionTest.php]:
```php
class Answers {
 /* ... */
    public static function fromArray(array $answersData) : Answers
    {
        $answers = [];
        try {
            foreach ($answersData as $answerData) {
                $answers[] = new Answer($answerData[0], $answerData[1]);
            }
        }catch(\Exception $e) {
             InvalidAnswerData::create();
        }
        
        return new self($answers);
    }
}
````

Comment (mickaelandrieu) [inline tests/Certificationy/QuestionTest.php]: was bad idea, we don't want to create hard bound between `Answers` utility class and the implementation, the Collection should only depends on the interface.

Moved to `TestCase` class instead.

**Comment by mickaelandrieu** [comment]
==> and v2 is here <==

**Pull request merged**